File: app/src/main/java/com/site/vs/videostation/adapter/base/RecyclerDataOperate.java
package com.site.vs.videostation.adapter.base;

import java.util.List;
import java.util.Set;

/**
 * 数据操作接口
 *
 * @author zhangbb
 * @date 2016/10/19
 */
public interface RecyclerDataOperate<T> {

    void add(T elem);

    void add(T elem, int index);

    void addAll(List<T> elem);

    void addAll(Set<T> elem);

    void replaceAll(List<T> elem);

    void remove(T elem);

    void removeAll();

}

File: app/src/main/java/com/site/vs/videostation/adapter/base/RecyclerDataOperate.kt
package com.site.vs.videostation.adapter.base

/**
 * 数据操作接口
 *
 * @author zhangbb
 * @date 2016/10/19
 */
interface RecyclerDataOperate<T> {

    fun add(elem: T)

    fun add(elem: T, index: Int)

    fun addAll(elem: List<T>)

    fun addAll(elem: Set<T>)

    fun replaceAll(elem: List<T>)

    fun remove(elem: T)

    fun removeAll()

}
